interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is a place holder that a function uses for any given argument. The argument is the data itself being fed to the function.
 
-Researched answer:
+Researched answer: A parameter is a variable used as an alias that is defined in the parenthesis of a function's syntax. An argument is the actual content used when invoking the function.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: It takes value, index, and array. Any one or number of them can be used in the function, but they have to be given as parameters in that order.
 
-Researched answer:
+Researched answer: The .map built in method takes element, index, and array as predefined parameters, in that order. Only element is required for the function. The parameter names are arbitrary, but the order they are declared in the function is not.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: both iterate through a given array, but .map returns an array with an equal number of elements while filter removes items from the given array, returning a smaller array.
 
-Researched answer:
+Researched answer: map will perform its inner function on each item in the array as it iterates, and return a new array with the same number of elements. Filter uses the inner function to decide if each element is true or false, and returns an array of only given elements it found true.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Repeating the same process until a certain condition is met.
```

Comment:
To expand, talk about some of the different forms of iteration you have used in your code.  Ex: I have used for loops, and higher order functions like map and filter to iterate over arrays.